core/unit_test/TestDeviceAndThreads.py

```diff
@@ -30,7 +31,20 @@ def GetFlag(flag, *extra_args):
     return int(p.stdout)
 
 def GetNumThreads(max_threads):
-    for x in [1, 2, 3, 5, 7]:
+    args = []
+    name = platform.system()
+    if name == 'Darwin':
+        args = ['sysctl', '-n', 'hw.physicalcpu_max']
+    elif name == 'Linux':
+        args = ['nproc', '--all']
+    else:
+        args = ['wmic', 'cpu', 'get', 'NumberOfCores']
+
+    result = subprocess.run(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    output = result.stdout.decode('utf-8')
+    phys_cores_count = int(output)
+    looplist = [1] + [i*phys_cores_count for i in [1,2,3,4,5,6,7]]
```

Comment:
Does that mean for 2 cores you test 1, 2, 4, 6 etc?

I explicitly want to test uneven number of threads at least on our nightly platforms maybe we should just do a loop from 1 to min(core_count, 8) or so?

Reply:
Yes that is what the loop does. 
But for numcores = 3 we get odd numbers too.  

If we test odd numbers of threads if numcores = 2, the test triggers the imbalance.

Reply:
Also, it seems to me this test aims at testing that what is passed via num-threads actually matches the hat is used internally, so why is my solution not general enough?

Reply:
I think we should just make sure that we test with an uneven number of threads in a build that doesn't use `hwloc`.

Reply:
how about this solution: https://github.com/kokkos/kokkos/pull/6604/commits/374064ab75aa4fd1727781b01c8c1724470788ac?